frontend/src/hooks/useProducts.js

```diff
@@ -0,0 +1,36 @@
+import { useState, useEffect } from 'react'
+import { fetchProducts } from '../communications/userApi'
+import {
+  removeAccentsFromWords,
+  highlightTableProps,
+} from '../components/frequentlyUsedFunctionsAndVariables'
+
+const useProducts = (searchKeyword) => {
+  const [listOfAllProducts, setListOfAllProducts] = useState([])
+  const [products, setProducts] = useState([])
+
+  const getAllProducts = async () => {
+    const fetchedProducts = await fetchProducts()
+    setListOfAllProducts(fetchedProducts)
+  }
+
+  useEffect(() => {
+    getAllProducts()
+  }, [])
+
+  useEffect(() => {
+    const searchWord = removeAccentsFromWords(searchKeyword)
+    setProducts(
+      listOfAllProducts.filter((product) =>
+        Object.entries(product).filter(
+          ([key, value]) =>
+            highlightTableProps.includes(key) && value.includes(searchWord)
```

Comment:
Amúgy ha csak itt van ez a lista használva, akkor nem is biztos, hogy kell neki külön file, lehet a useProducts előtt egy nem exportált változó.

Reply:
Van a ListElement-ben is.